desc/descriptor.go

```diff
@@ -891,10 +892,29 @@ func (fd *FieldDescriptor) String() string {
 // GetJSONName returns the name of the field as referenced in the message's JSON
 // format.
 func (fd *FieldDescriptor) GetJSONName() string {
-	if jsonName := fd.proto.GetJsonName(); jsonName != "" {
-		return jsonName
+	if jsonName := fd.proto.JsonName; jsonName != nil {
+		// if json name is present, use its value
+		return *jsonName
 	}
-	return fd.proto.GetName()
+	// otherwise, compute the proper JSON name from the field name
+	return jsonCamelCase(fd.proto.GetName())
+}
+
+func jsonCamelCase(s string) string {
```

Comment:
Is there a link to the code within github.com/protocolbuffers/protobuf where this logic is done, and could we add this link as a comment here, to make sure it's understood what this is matching to?

I ask because there's often disagreement as to what constitutes camel case, especially around numbers and abbrevations, and regardless of what is "correct", it's important to match `protoc` here.

I assume this does match up, but IMO it's important to document for the future in case someone looks at this code and wonders the same.